### ISSUE TITLE
bad-super-call false positive w/ YES message.

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#280]
Originally reported by: **the mulhern (BitBucket: [the_mulhern](http://bitbucket.org/the_mulhern))**

---

Example:

```
#!python

class ClassFactory(object):

    def makeClass():

        def __init__(self):
            super(self.__class__, self).__init__()

        return type(
           "Class",
           (object,),
           {"__init__": __init__})

```

yields error:

E:  6,12: Bad first argument YES given to super() (bad-super-call)

I have no idea where the YES came from.

The following code:

```
#!python

class Class(object):

    def __init__(self):
        super(self.__class__, self).__init__()
```

yields no such error, no bad-super-call error at all.

---
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/280
